library/ssl_tls.c

```diff
@@ -1352,19 +1352,29 @@ void mbedtls_ssl_conf_session_cache( mbedtls_ssl_config *conf,
 int mbedtls_ssl_set_session( mbedtls_ssl_context *ssl, const mbedtls_ssl_session *session )
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-
+    mbedtls_ssl_session *target = ssl->session_negotiate;
     if( ssl == NULL ||
         session == NULL ||
         ssl->session_negotiate == NULL ||
         ssl->conf->endpoint != MBEDTLS_SSL_IS_CLIENT )
     {
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
-
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
+    if( ssl->handshake->resume == 1 )
+    {
+        target = mbedtls_calloc( 1, sizeof(mbedtls_ssl_session) );
```

Comment:
I think we should check that `ssl->session_negotiate->tls_version == MBEDTLS_SSL_VERSION_TLS1_3` and `session->tls_version == MBEDTLS_SSL_VERSION_TLS1_3`. If this is not the case we return with `MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE`.